### PR TITLE
UcrGeomListWithAes data type restrictions for aes receiver bug (+ucrReceiver and ucrGeom)

### PR DESCRIPTION
**Pull request by FrancoisJRenaud**
in progress...

### CODE REVIEW

**Comment by dannyparsons** [comment]
This is enough for one pull request.

**Comment by FrancoisJRenaud** [comment]
Testing went fine. Everything is working well now. As strExcludeDataTypes is not used at all in ucrGeom for the meoment, I recommend for testing to change a random aes parameter and set strExcludeDataTypes to smthgsmthg. By doing so, realised that the values inside strExcludeDataTypes where changed during the process : indeed, in the SetExcludedDataTypes in ucrReceiver, the parameter strExclude := clsCurrGeom.clsAesParameters(i).strExcludedDataTypes was modified by adding quotes, which was directly performed on clsCurrGeom.clsAesParameters(i).strExcludedDataTypes. Now added the procedure that was there already in the SetIncludedDataTypes. 
Then also added the temporary procedure for including ordered,factors in the exclude case.

File: instat/ucrReceiver.vb [hunk 1]
```diff
@@ -143,17 +143,22 @@ Public Class ucrReceiver
...
         Array.Copy(strInclude, strTypes, strInclude.Length)
+        'If the two previous lines where not added, the modification of value performed on strTypes was immediately performed on strInclude, then the argument passed into the function such as clsCurrGeom.clsAesParameters(i).strIncludedDataTypes in ucrGeomListWithAes.SetParameters would have been edited (i.e. quotes would have been added to the types names in the strIncludedDataTypes of the i'th AesParameter of the current Geom...), which we don't want !
         For i = 0 To strInclude.Count - 1
             strTypes(i) = Chr(34) & strInclude(i) & Chr(34)
         Next
         AddIncludedMetadataProperty("class", strTypes)
     End Sub
 
     Public Sub SetExcludedDataTypes(strExclude As String())
+        Dim strTypes(strExclude.Count - 1) As String
```
Comment (FrancoisJRenaud) [inline instat/ucrReceiver.vb]: realised that the values inside strExcludeDataTypes where changed during the process : indeed, in the SetExcludedDataTypes in ucrReceiver, the parameter strExclude := clsCurrGeom.clsAesParameters(i).strExcludedDataTypes was modified by adding quotes, which was directly performed on clsCurrGeom.clsAesParameters(i).strExcludedDataTypes. Now added the procedure that was there already in the SetIncludedDataTypes.

File: instat/ucrReceiver.vb [hunk 2]
```diff
@@ -216,7 +221,13 @@ Public Class ucrReceiver
         Dim iExcludeIndex As Integer
 
         Dim kvpExcludeProperty As KeyValuePair(Of String, String())
-
+        If strProperty = "class" AndAlso strExclude.Contains(Chr(34) & "factor" & Chr(34)) Then
```
Comment (FrancoisJRenaud) [inline instat/ucrReceiver.vb]: Then also added the temporary procedure for including ordered,factors in the exclude case.

Comment (FrancoisJRenaud) [inline instat/ucrReceiver.vb]: @dannyparsons Checked on the pull request. The restriction on the data types of the aesthetics receivers is now working fine.
steven

**Pull request merged**